src/main/java/sg/edu/sutd/bank/webapp/service/ClientAccountDAOImpl.java

```diff
@@ -61,5 +65,57 @@ public void update(ClientAccount clientAccount) throws ServiceException {
 			closeDb(conn, ps, rs);
 		}
 	}
+	
+	/* get transaction details: amount, sender, receiver from client_transaction
+	   update sender and receiver alike
+	*/
+	public void executeTransaction(List<ClientTransaction> transactions) throws ServiceException {	
+		Connection conn = connectDB();
+		try {
+			for (ClientTransaction transaction : transactions) {
+				if (transaction.getStatus() == TransactionStatus.APPROVED) {
+					PreparedStatement ps = prepareStmt(conn, "SELECT * FROM client_transaction WHERE id = ?");
+					ps.setInt(1, transaction.getId()); // transaction-specific id
+					ResultSet rs = ps.executeQuery();
+					rs.next();
+					
+					// extracts USERID of sender & receiver
+					transaction.setFromAccount(rs.getInt("user_id"));
+					transaction.setToAccount(rs.getInt("to_account_num"));
+					BigDecimal amount = rs.getBigDecimal("amount");
+					
+					// Deduct money from sender's account
+					PreparedStatement psf = prepareStmt(conn, "UPDATE client_account SET amount = amount - ? WHERE user_id = ? AND amount >= ?");
```

Comment:
Why need to check if amount >= amount here? If its to ensure that the sender has enough money in his account, why don't we do something like:
1. check how much money sender has now
2. if sender has insufficient money to send the transfer, throw error
3. otherwise:
```
PreparedStatement psf = prepareStmt(conn, "UPDATE client_account SET amount = ? WHERE user_id = ?");
pst.setBigDecimal(1, newAmount);
pst.setInt(2, transaction.getToAccount());
```

Reply:
okay, im gonna shift the checks up to the servlet level instead. can do all the checks there instead of at the transaction-decision level